docs/os-ref.md

```diff
@@ -4,615 +4,546 @@ title: aragonOS reference documentation
-the entity has permission to create a new version.
+![forwarding animation](/docs/assets/fwd.gif)
+(Governance model and characters are fictional)
 
-After the ACL check, the Repo logic checks whether the version upgrade is allowed.
-A version bump for a package is defined by the following rules:
+### EVMScripts
 
-- Only one member of the version is increased by 1. The version components to the
-left of the raised member must stay the same and the components to the right must be 0.
-  - Example: From `2.1.3` the only allowed bumps are to `3.0.0` (major version),
-  `2.2.0` (minor version), and `2.1.4` (patch version).
-- Changes to the app code address can only be done if the raise changes the major
-version (upgrading it to `M.0.0` by the above rule).
+We have designed our own scripting format, known as EVMScripts, to encode complex actions into a bytes representation that can be stored and later executed by another entity. EVMScripts can be installed on a per-organization basis, through a **EVMScriptRegistry**, and aragonOS comes complete with the ability to install multiple script executors in an organization.
 
-The initial version of an app must be a valid bump from version `0.0.0`.
+**EVMScript executors** are contracts that take a script and an input and return an output after execution.
 
-By having this check performed at the smart contract level, we can load the correct
-version of the frontend just by looking at an instance of an app. This is done by
-checking that the version of a smart contract is linked to a given app by getting
-its `appId` and `appCode` (See section *6.3.2.3 By latest contract address*)
+**EVMScript executors and EVMScriptRegistry**
 
-#### 6.3.2 Fetching Repo versions
-Repos offer multiple ways to fetch versions. By checking logs for the following
-event one can see all the versions ever created in a Repo:
+EVMScript executors must implement the following interface:
 
 ```solidity
-(Repo) NewVersion(uint256 versionId, uint16[3] semanticVersion);
+interface IEVMScriptExecutor {
+    function execScript(bytes script, bytes input, address[] blacklist) external returns (bytes);
+    function executorType() external pure returns (bytes32);
+}
 ```
 
-All different methods for fetching versions return the following tuple:
+> **Warning**
+>
+> EVMScript executors are called with a `delegatecall`, and operate in the context of the calling app. This **must** be taken into consideration when developing your own executor, as it could cause a security breach.
 
-```solidity
-repoVersion = (uint16[3] semanticVersion, address contractAddress, bytes contentURI)
-```
+aragonOS provides the `CallsScript` executor as a simple way to concatenate multiple calls. It cancels the operation if any of the calls fail.
 
-#### 6.3.2.1 By versionId
-Every version can be fetched with its `versionId` (which starts in `1` and is
-increments by `1` each version).
+- **Script body:** (See source code file for spec of the payload)
```

Comment:
which source code file is this referring to?  can we link to it?

Reply:
Fixed in 63c5542